models/points.go

```diff
@@ -89,6 +89,33 @@ type Point interface {
 	// AppendString appends the result of String() to the provided buffer and returns
 	// the result, potentially reducing string allocations
 	AppendString(buf []byte) []byte
+
+	// FieldIterator retuns a FieldIterator that can be used to traverse the
+	// fields of a point without constructing the in-memory map
+	FieldIterator() FieldIterator
+}
+
+type FieldType int
+
+const (
+	Integer FieldType = iota
+	Float
+	Boolean
+	String
+	Empty
+)
+
+type FieldIterator interface {
+	Next() bool
+	FieldKey() []byte
```

Comment:
`Point` implement this interface and already has `Key` defined which would conflict with intent `FieldKey`